src/service/performance-impl.js

```diff
@@ -248,6 +299,53 @@ export class Performance {
     });
   }
 
+  /**
+   * When the visibility state of the document changes to hidden,
+   * send the layout jank score.
+   */
+  onVisibilityChange_() {
+    if (this.win.document.visibilityState !== 'hidden') {
+      return;
+    }
+    this.tickLayoutJankScore_();
+  }
+
+  /**
+   * Tick the layout jank score metric.
+   *
+   * A value of the metric is recorded in under two names, `lj` and `lj-2`,
+   * for the first two times the page transitions into a hidden lifecycle state
+   * (when the page is navigated a way from, the tab is backgrounded for
+   * another tab, or the user backgrounds the browser application).
+   *
+   * Since we can't reliably detect when a page session finally ends,
+   * recording the value for these first two events should provide a fair
+   * amount of visibility into this metric.
+   */
+  tickLayoutJankScore_() {
+    if (!this.tickedFirstJankScore_) {
+      this.tickDelta('lj', this.aggregateJankScore_);
+      this.flush();
+      this.tickedFirstJankScore_ = true;
+      return;
+    }
+    if (!this.tickedSecondJankScore_) {
+      this.tickDelta('lj-2', this.aggregateJankScore_);
+      this.flush();
+      this.tickedSecondJankScore_ = true;
+
+      this.win.removeEventListener(
+          'visibilitychange',
+          this.onVisibilityChange_,
+          {capture: true}
+      );
+      // const platform = Services.platformFor(this.win);
```

Comment:
If I uncomment this code our tests fail, I assume because the platform service hasn't had a chance to load yet...and suggestions here?

```
Error: Expected service platform to be registered
    at Log.assert (usr/local/google/home/ericnyc/git/amphtml/src/log.js:343:17)
    at devAssert (usr/local/google/home/ericnyc/git/amphtml/src/log.js:728:29)
    at getServiceInternal (usr/local/google/home/ericnyc/git/amphtml/src/service.js:394:3)
    at getService (usr/local/google/home/ericnyc/git/amphtml/src/service.js:194:10)
    at Function.platformFor (usr/local/google/home/ericnyc/git/amphtml/src/services.js:317:7)
    at Performance.tickLayoutJankScore_ (usr/local/google/home/ericnyc/git/amphtml/src/service/performance-impl.js:342:33)
    at Performance.onVisibilityChange_ (usr/local/google/home/ericnyc/git/amphtml/src/service/performance-impl.js:310:10)
    at http://localhost:9876/usr/local/google/home/ericnyc/git/amphtml/test/unit/test-performance.js:1014:35
    at Array.forEach (<anonymous>)
    at Context.<anonymous> (usr/local/google/home/ericnyc/git/amphtml/test/unit/test-performance.js:1014:15)
```

Reply:
you'd have to make this call in `coreServicesAvailable` for it to be safe i believe. we could just duplicate the `isSafari` implementation here somewhere if that's the only thing you need

Reply:
I duplicated the isSafari() checks in this file to avoid the cross-service dependency problem issue